packages/snaps-execution-environments/webpack.config.js

```diff
@@ -0,0 +1,296 @@
+
+            return {
+              inlineSource: bundle.source(),
+              lockdownScript: SES_BUNDLE,
+            };
+          },
+        }),
+      ],
+    },
+  },
+
+  {
+    name: 'worker-executor',
+    entry: './src/webworker/executor/index.ts',
+    inline: true,
+
+    config: {
+      target: 'webworker',
+    },
+  },
+
+  {
+    name: 'worker-pool',
+    entry: './src/webworker/pool/index.ts',
+    scuttleGlobalThis: true,
+
+    config: DEFAULT_WEB_CONFIG,
+  },
+];
+
+/**
+ * The base Webpack configuration.
+ *
+ * @type {Configuration}
+ */
+const baseConfig = {
+  // We always set this to `production`, since the development build doesn't
+  // have much value, and it makes the build much larger.
+  mode: 'production',
+
+  output: {
+    filename: 'bundle.js',
+  },
+
+  module: {
+    rules: [
+      {
+        test: /\.tsx?$/u,
+        use: {
+          // While `ts-loader` isn't the fastest, it seems to produce smaller
+          // bundles than `swc-loader`.
+          loader: 'ts-loader',
+          options: {
+            transpileOnly: true,
+            configFile: resolve(__dirname, 'tsconfig.build.json'),
+            compilerOptions: {
+              module: 'esnext',
+              moduleResolution: 'bundler',
+              target: 'es2022',
+            },
+          },
+        },
+      },
+    ],
+  },
+
+  plugins: [
+    new ProvidePlugin({
+      process: 'process/browser',
+    }),
+  ],
+
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js'],
+    fallback: {
+      // Used in the `crypto` endowment, but only as fallback, so we can
+      // safely ignore it if it's not available.
+      crypto: false,
+
+      stream: require.resolve('stream-browserify'),
+    },
+
+    plugins: [
+      // `TsconfigPathsPlugin` causes issues with tree shaking, so we only
+      // enable it in development mode.
+      IS_PRODUCTION
+        ? null
+        : new TsconfigPathsPlugin({
+            configFile: resolve(__dirname, 'tsconfig.build.json'),
+            baseUrl: __dirname,
+          }),
+    ],
+  },
+};
+
+/**
+ * The Webpack configurations for each entry point.
+ *
+ * @type {Configuration[]}
+ */
+const configs = ENTRY_POINTS.map(
+  ({
+    name,
+    entry,
+    inline,
+    inlineBundle,
+    scuttleGlobalThis,
+    scuttleGlobalThisExceptions = [],
+    config = {},
+  }) =>
+    merge(baseConfig, config, {
+      name,
+      entry,
+
+      output: {
+        path: resolve(__dirname, 'dist/webpack', name),
+      },
+
+      plugins: [
+        new LavaMoatPlugin({
+          isBuiltin,
+          generatePolicy: UPDATE_POLICY,
+          policyLocation: resolve(__dirname, 'lavamoat', 'webpack', name),
+          inlineLockdown: inline ? /bundle\.js/u : undefined,
+          scuttleGlobalThis: {
+            enabled: scuttleGlobalThis,
+            exceptions: [
+              'Object',
+              'postMessage',
+              'Reflect',
+              'Set',
```

Comment:
`Set`, `Object` and `Reflect` are new exceptions, do we know why?

Reply:
Yes, these are introduced by LavaMoat boilerplate IIRC. These are safe to add, and we may be able to remove them with a future LavaMoat update.